pkg/metrics/sketch_series.go

```diff
@@ -106,98 +107,198 @@ func (sl SketchSeriesList) MarshalJSON() ([]byte, error) {
 // it's contents are marshaled individually, packed with the appropriate protobuf metadata, and compressed in stream.
 // The resulting payloads (when decompressed) are binary equal to the result of marshaling the whole object at once.
 func (sl SketchSeriesList) MarshalSplitCompress(bufferContext *marshaler.BufferContext) ([]*[]byte, error) {
-	// The Metadata field of gogen.SketchPayload is never written to - so pack an empty metadata as the footer
-	footer := []byte{0x12, 0}
+	var err error
+	var compressor *stream.Compressor
+	buf := bufferContext.PrecompressionBuf
+	ps := molecule.NewProtoStream(buf)
+	payloads := []*[]byte{}
+
+	// constants for the protobuf data we will be writing, taken from
+	// https://github.com/DataDog/agent-payload/blob/a2cd634bc9c088865b75c6410335270e6d780416/proto/metrics/agent_payload.proto#L47-L81
+	const payloadSketches = 1
+	const payloadMetadata = 2
+	const sketchMetric = 1
+	const sketchHost = 2
+	const sketchDistributions = 3
+	const sketchTags = 4
+	const sketchDogsketches = 7
+	/* unused
+	const distributionTs = 1
+	const distributionCnt = 2
+	const distributionMin = 3
+	const distributionMax = 4
+	const distributionAvg = 5
+	const distributionSum = 6
+	const distributionV = 7
+	const distributionG = 8
+	const distributionDelta = 9
+	const distributionBuf = 10
+	*/
+	const dogsketchTs = 1
+	const dogsketchCnt = 2
+	const dogsketchMin = 3
+	const dogsketchMax = 4
+	const dogsketchAvg = 5
+	const dogsketchSum = 6
+	const dogsketchK = 7
+	const dogsketchN = 8
+
+	// generate a footer containing an empty Metadata field (TODO: this isn't
```

Comment:
Is this `TODO` addressable now? or is this a future task

Reply:
I could -- I left it in to produce the same binary output.  Do you agree that it can be removed?

Reply:
I see.
Personally i'd keep it to maintain 1:1 binary compatibility (and maybe change the comment slightly to note that). But that's just my opinion.

Reply:
Ah, I removed it in the last commit.  Let's land it as-is.  In the unlikely event this causes issues, they should be quite obvious (sketches not appearing on the app).

Reply:
I've changed my mind after having a look through the generated marshaling code.  It's pretty intentionally including the empty Metadata message, instead of omitting it entirely.